lib/main.c

```diff
@@ -451,4 +453,201 @@ int ec_seckey_verify(const unsigned char *seckey)
   int ret = secp256k1_ec_seckey_verify(ctx, seckey);
   secp256k1_context_destroy(ctx);
   return ret;
-}
\ No newline at end of file
+}
+
+void **alloc_pointer_arr(size_t n, size_t elem_size)
+{
+  void **arr = malloc(sizeof(void*) * n);
+  for (int i = 0; i < n; i++)
+  {
+    arr[i] = malloc(elem_size);
+  }
+  return arr;
+}
+
+void free_pointer_arr(void **ptrs, size_t n)
+{
+  for (int i = 0; i < n; i++)
+  {
+    free(ptrs[i]);
+  }
+  free(ptrs);
+}
+
+#define RETURN_ON_ZERO              \
+  if (ret == 0)                     \
+  {                                 \
+    secp256k1_context_destroy(ctx); \
+    return ret;                     \
+  }
```

Comment:
not an expert of C, just to know, which `ctx` is destroyed by this macro?

Reply:
Depends on where you use it. Macros in C are just textual replacement. So whatever `ctx` is in scope when using the macro will be destroyed